### PR TITLE
Add non-infix parseField, also explicit liftParseField

### PR DESCRIPTION
**Pull request by phadej**

### CODE REVIEW

File: Data/Aeson/Types/FromJSON.hs [hunk 1]
```diff
@@ -59,11 +59,18 @@ module Data.Aeson.Types.FromJSON
     -- * Functions
     , fromJSON
     , ifromJSON
+    , typeMismatch
+    , parseField
+    , parseFieldMaybe
+    , parseFieldMaybe'
+    , liftParseField
```
Comment (phadej) [inline Data/Aeson/Types/FromJSON.hs]: I'm not sure `lift`but it kind of make sense here.

**Comment by phadej** [comment]
I'm not sure whether these functions could be exported from `Data.Aeson`

File: Data/Aeson/Types/FromJSON.hs (hunk 1 above)
Comment (bergmark) [inline Data/Aeson/Types/FromJSON.hs]: opaleye uses `explicit` for this

**Comment by bergmark** [comment]
A pro of using operator aliases is that the operators get a pronouncable name. Cons are that there are twice as many operator names to memorize, and that TMTOWTDI. I'm on the fence about this.

**Comment by bergmark** [comment]
... another pro is that partial application becomes more legible

**Comment by phadej** [comment]
This is motivated by the need for `liftParseField`, as the haddock points out the example usage.

**Comment by phadej** [comment]
Renamed to `explicitParseField`

**Comment by bergmark** [comment]
Thanks!

**Pull request merged**